packages/grafana-ui/src/components/SelectNG/hooks.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useCallback, useState, useRef, useEffect } from 'react';
+import { SelectableValue } from '@grafana/data';
+import { flattenOptions, minSameWidthModifier, shouldAllowOptionCreate } from './utils';
+import { usePopper } from 'react-popper';
+import { Placement } from '@popperjs/core';
+import { AsyncSelectOptionsResolver } from './types';
+import debounce from 'debounce-promise';
+
+export interface UseSelectOptions<T> {
+  placement: Placement;
+}
+
+export interface UseAsyncSelectOptions<T> extends Omit<UseSelectOptions<T>, 'options'> {
+  loadOptions: AsyncSelectOptionsResolver<T>;
+}
+
+// Util hook for proiv
+export function useSelect<T = any>({ placement }: UseSelectOptions<T>) {
+  const triggerRef = React.createRef<HTMLInputElement>();
+  const [popperRef, setPopperRef] = useState<HTMLDivElement | null>(null);
+  const [popperElement, setPopperElement] = useState<HTMLDivElement | null>(null);
+
+  const popper = usePopper(popperRef, popperElement, {
+    modifiers: [
+      minSameWidthModifier as any,
+      {
+        name: 'arrow',
+        options: {
+          enabled: false,
+        },
+      },
+      {
+        name: 'flip',
+        options: {
+          enabled: false,
+        },
+      },
+    ],
+    placement,
+  });
+
+  const popperProps = {
+    style: popper.styles.popper,
+    ...popper.attributes.popper,
+  };
+
+  const defaultFilter = (o: SelectableValue, inputValue: string) => {
+    return o.label!.includes(inputValue);
+  };
+  // naive filter implementation, will probably need customization
+  const filterOptions = (filterable: SelectableValue[], inputValue?: string | null) => {
```

Comment:
Maybe we could use [match-sorter](https://github.com/kentcdodds/match-sorter) for this?